src/data/real/basic.lean

```diff
@@ -378,35 +390,21 @@ theorem exists_sup (S : set ℝ) : (∃ x, x ∈ S) → (∃ x, ∀ y ∈ S, y 
     refine le_trans _ (le_of_lt $ hf₂ _ n0 _ xS),
     rwa [le_sub, inv_le ((nat.cast_pos.2 n0):((_:ℝ) < _)) xz] },
   { exact mk_le_of_forall_le ⟨1, λ n n1,
-      let ⟨x, xS, hx⟩ := hf₁ _ n1 in le_trans hx (h _ xS)⟩ },
-  intros ε ε0,
-  suffices : ∀ j k ≥ nat_ceil ε⁻¹, (f j / j - f k / k : ℚ) < ε,
-  { refine ⟨_, λ j ij, abs_lt.2 ⟨_, this _ _ ij (le_refl _)⟩⟩,
-    rw [neg_lt, neg_sub], exact this _ _ (le_refl _) ij },
-  intros j k ij ik,
-  replace ij := le_trans (le_nat_ceil _) (nat.cast_le.2 ij),
-  replace ik := le_trans (le_nat_ceil _) (nat.cast_le.2 ik),
-  have j0 := nat.cast_pos.1 (lt_of_lt_of_le (inv_pos.2 ε0) ij),
-  have k0 := nat.cast_pos.1 (lt_of_lt_of_le (inv_pos.2 ε0) ik),
-  rcases hf₁ _ j0 with ⟨y, yS, hy⟩,
-  refine lt_of_lt_of_le ((@rat.cast_lt ℝ _ _ _).1 _)
-    ((inv_le ε0 (nat.cast_pos.2 k0)).1 ik),
-  simpa using sub_lt_iff_lt_add'.2
-    (lt_of_le_of_lt hy $ sub_lt_iff_lt_add.1 $ hf₂ _ k0 _ yS)
+      let ⟨x, xS, hx⟩ := hf₁ _ n1 in le_trans hx (h xS)⟩ }
 end
 
 noncomputable instance : has_Sup ℝ :=
 ⟨λ S, if h : (∃ x, x ∈ S) ∧ (∃ x, ∀ y ∈ S, y ≤ x)
```

Comment:
Is it worth replacing this with the named props too?
```suggestion
⟨λ S, if h : S.nonempty ∧ bdd_above S
```

Reply:
I'm going to generalize this constructor, so these lines will go away soon.